client-spark/spark-3/src/main/scala/org/apache/spark/shuffle/celeborn/CelebornShuffleReader.scala

```diff
@@ -124,12 +146,31 @@ class CelebornShuffleReader[K, C](
           TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startFetchWait))
         // ensure inputStream is closed when task completes
         context.addTaskCompletionListener[Unit](_ => inputStream.close())
-        inputStream
+        (partitionId, inputStream)
       } else {
-        CelebornInputStream.empty()
+        (partitionId, CelebornInputStream.empty())
       }
-    }).flatMap(
-      serializerInstance.deserializeStream(_).asKeyValueIterator)
+    }).map { case (partitionId, inputStream) =>
+      (partitionId, serializerInstance.deserializeStream(inputStream).asKeyValueIterator)
+    }.flatMap { case (partitionId, iter) =>
+      try {
+        iter
+      } catch {
+        case e: CelebornIOException =>
```

Comment:
How to handle other kind of Exception?

Reply:
I've discussed with @waitinfuture previously, and was told most exception is converted to CelebornIOException before throwing it.
It should be safe in current implementation